rotkehlchen/chain/manager.py

```diff
@@ -824,384 +829,49 @@ def sync_bitcoin_accounts_with_db(
-        """
-        bch_usd_price = Inquirer().find_usd_price(A_BCH)
-        remove_with_populated_balance = (
-            append_or_remove == 'remove' and len(self.balances.bch) != 0
-        )
-        # Query the balance of the account except for the case when it's removed
-        # and there is no other account in the balances
-        if append_or_remove == 'append' or remove_with_populated_balance:
-            if already_queried_balance is None:
-                balances = get_bitcoin_cash_addresses_balances([account])
-                balance = balances[account]
-            else:
-                balance = already_queried_balance
-            usd_balance = balance * bch_usd_price
-
-        if append_or_remove == 'append':
-            self.balances.bch[account] = Balance(amount=balance, usd_value=usd_balance)
-        elif append_or_remove == 'remove':
-            if account in self.balances.bch:
-                del self.balances.bch[account]
-        else:
-            raise AssertionError('Programmer error: Should be append or remove')
-
-        if len(self.balances.bch) == 0:
-            # If the last account was removed balance should be 0
-            self.totals.assets[A_BCH] = Balance()
-        else:
-            self.totals.assets[A_BCH] = add_or_sub(
-                self.totals.assets[A_BCH],
-                Balance(balance, usd_balance),
-            )
-
-        # At the very end add/remove it from the accounts
-        getattr(self.accounts.bch, append_or_remove)(account)
-
-    def modify_eth_account(
-            self,
-            account: ChecksumEthAddress,
-            append_or_remove: str,
-    ) -> None:
-        """Either appends or removes an ETH acccount.
-
-        Call with 'append' to add the account
-        Call with 'remove' remove the account
-
-        May raise:
-        - Input error if the given_account is not a valid ETH address
-        - BadFunctionCallOutput if a token is queried from a local chain
-        and the chain is not synced
-        - RemoteError if there is a problem with a query to an external
-        service such as Etherscan or cryptocompare
-        """
-        eth_usd_price = Inquirer().find_usd_price(A_ETH)
-        remove_with_populated_balance = (
-            append_or_remove == 'remove' and len(self.balances.eth) != 0
-        )
-        # Query the balance of the account except for the case when it's removed
-        # and there is no other account in the balances
-        if append_or_remove == 'append' or remove_with_populated_balance:
-            amount = self.ethereum.get_eth_balance(account)
-            usd_value = amount * eth_usd_price
-
-        if append_or_remove == 'append':
-            self.accounts.eth.append(account)
-            self.balances.eth[account] = BalanceSheet(
-                assets=defaultdict(Balance, {A_ETH: Balance(amount, usd_value)}),
-            )
-        elif append_or_remove == 'remove':
```

Comment:
you need to keep this and generally all parts of the removal balance (and remember -- for all chains, not only ETH)

Reply:
Done (not pushed yet)